File: LC1768.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        m = len(word1)
        n = len(word2)
        s = ""
        i = 0
        j = 0

        while i < m or j < n:
            if i < m:
                s += word1[i]
                i += 1
            
            if j < n:
                s += word2[j]
                j += 1
        
        return s

File: LC1768.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string s = "";

        int i = 0, j = 0;
        int m = word1.size(), n = word2.size();

        while(i < m || j < n){
            if(i < m)
                s.push_back(word1[i++]);
            if(j < n)
                s.push_back(word2[j++]);
        }
        
        return s;
    }
};
